src/app/page.js

```diff
@@ -21,10 +21,23 @@ const sortOptions = [
 
 export default function Page() {
   const params = useSearchParams();
+  const router = useRouter();
   const alert = params.get("alert");
   const message = params.get("message");
   const [keyword, setKeyword] = useState("");
   const [sort, setSort] = useState(sortOptions[0].value);
+  const [showAlert, setShowAlert] = useState(false);
+
+  useEffect(() => {
+    if (alert) {
+      setShowAlert(true);
+      setTimeout(() => {
+        setShowAlert(false);
+        // Removed alert because it can re-render the Toast Component
+        router.push(`/?done=success&message=${message}`, { scroll: false });
```

Comment:
As mentioned in a previous inline comment, I think this should be `/`?

Reply:
changed router path `/?done=success&message=${message}` to `/`

Reply:
I think this is correct otherwise the alert will keep appearing? Did you test it?

Reply:
Yes, tested it.

Reply:
I tested it many times until I finally found the solution.

Reply:
> changed router path `/?done=success&message=${message}` to `/`
Then I found that this `/?done=success&message=${message}` will be a better way to stop re-rendering.